packages/integration-tests/tests/bridge_erc20.ts

```diff
@@ -248,112 +412,156 @@ describe('Bridging ERC20 tokens', async function () {
-      await fuel_bridge.functions
-        .asset_to_l1_address({ bits: fuel_testAssetId })
-        .addContracts([fuel_bridge, fuel_bridgeImpl])
-        .call();
-
-      const { value: l2_decimals } = await fuel_bridge.functions
-        .decimals({ bits: fuel_testAssetId })
-        .addContracts([fuel_bridge, fuel_bridgeImpl])
-        .get();
-
-      expect(l2_decimals).to.be.equal(9);
+      for (let i = 0; i < fuelAssetId.length; i++) {
+        await fuel_bridge.functions
+          .asset_to_l1_address({ bits: fuelAssetId[i] })
+          .addContracts([fuel_bridge, fuel_bridgeImpl])
+          .dryRun();
+        const { value: l2_decimals } = await fuel_bridge.functions
+          .decimals({ bits: fuelAssetId[i] })
+          .addContracts([fuel_bridge, fuel_bridgeImpl])
+          .get();
+
+        if (i == fuelAssetId.length - 1) {
+          expect(l2_decimals).to.be.equal(9);
+        } else {
+          expect(l2_decimals.toString()).to.be.equal(decimals[i].toString());
+        }
+      }
     });
 
     it('Check ERC20 arrived on Fuel', async () => {
-      // check that the recipient balance has increased by the expected amount
-      const newReceiverBalance = await fuelTokenReceiver.getBalance(
-        fuel_testAssetId
-      );
-
-      expect(
-        newReceiverBalance.eq(
-          fuelTokenReceiverBalance.add(toBeHex(NUM_TOKENS / DECIMAL_DIFF))
-        )
-      ).to.be.true;
+      for (let i = 0; i < fuelAssetId.length; i++) {
+        // check that the recipient balance has increased by the expected amount
+        const newReceiverBalance = await fuelTokenReceiver.getBalance(
+          fuelAssetId[i]
+        );
+        if (i == fuelAssetId.length - 1) {
+          expect(
+            newReceiverBalance.eq(
+              fuelTokenReceiverBalance[i].add(
+                toBeHex(WETH_DEPOSIT / DECIMAL_DIFF)
+              )
+            )
+          ).to.be.true;
+        } else {
+          expect(
+            newReceiverBalance.eq(
+              fuelTokenReceiverBalance[i].add(
+                toBeHex(
+                  NUM_TOKENS /
+                    DECIMAL_DIFF /
+                    (DECIMAL_DIFF / 10n ** decimals[i])
+                )
+              )
+            )
+          ).to.be.true;
+        }
+      }
     });
 
     it('Bridge metadata', async () => {
-      // use the FuelERC20Gateway to deposit test tokens and receive equivalent tokens on Fuel
-      const receipt = await env.eth.fuelERC20Gateway
-        .connect(ethereumTokenSender)
-        .sendMetadata(eth_testTokenAddress)
-        .then((tx) => tx.wait());
-
-      // parse events from logs
-      const [event, ...restOfEvents] =
-        await env.eth.fuelMessagePortal.queryFilter(
-          env.eth.fuelMessagePortal.filters.MessageSent,
-          receipt.blockNumber,
-          receipt.blockNumber
+      // TODO: ADD WETH
+      for (let i = 0; i < fuelAssetId.length; i++) {
```

Comment:
All of these for loops make the testing logic more convoluted than it already is. Ideally avoid for loops inside `it` blocks. 

Try to use a testing table, take example of https://github.com/FuelLabs/fuel-bridge/blob/0b6e3b198707d9fd7000d255b0ad3164d7e64507/packages/solidity-contracts/test/behaviors/erc20GatewayV4.behavior.test.ts#L317